components/practice/learndl.js

```diff
@@ -248,6 +279,26 @@ class LearndlPractice extends RequesterMixin(Localizer(LitElement)) {
 			value = event.target.value;
 		}
 
+		if (event.target.classList.contains('learndl-answer')) {
+			if (this.activityData.answers.find(answer => answer.value.toLowerCase() === value.toLowerCase())) {
+				this.activityData.answers[prop.split(':')[1]].duplicate = true;
+				this._updateActivity();
+				return;
+			} else {
+				this.activityData.answers[prop.split(':')[1]].duplicate = false;
+			}
+		}
+		if (event.target.classList.contains('learndl-attribute')) {
+			if (this.activityData.attributes.find(attribute => attribute.value.toLowerCase() === value.toLowerCase())) {
+				event.target.setAttribute('aria-invalid', true);
+				this.activityData.attributes[prop.split(':')[1]].duplicate = true;
```

Comment:
if you split property at the start you could combine the duplicate checks
```suggestion
				this.activityData[propArr[0]][propArr[1]].duplicate = true;
```

Reply:
Good catch